new-installer/internal/steps/aws/kms_test.go

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: 2025 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package steps_aws_test
+
+import (
+	"crypto/rand"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-exec/tfexec"
+	"github.com/open-edge-platform/edge-manageability-framework/installer/internal"
+	"github.com/open-edge-platform/edge-manageability-framework/installer/internal/config"
+	"github.com/open-edge-platform/edge-manageability-framework/installer/internal/steps"
+	steps_aws "github.com/open-edge-platform/edge-manageability-framework/installer/internal/steps/aws"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+type KMSStepTest struct {
+	suite.Suite
+	config       config.OrchInstallerConfig
+	runtimeState config.OrchInstallerRuntimeState
+	step         *steps_aws.KMSStep
+	randomText   string
+	logDir       string
+	tfUtility    *MockTerraformUtility
+}
+
+func TestKMSStep(t *testing.T) {
+	suite.Run(t, new(KMSStepTest))
+}
+
+func (s *KMSStepTest) SetupTest() {
+	rootPath, err := filepath.Abs("../../../../")
+	s.Require().NoError(err, "Failed to get absolute path")
+	s.randomText = strings.ToLower(rand.Text()[0:8])
+	s.logDir = filepath.Join(rootPath, ".logs")
+	err = internal.InitLogger("debug", s.logDir)
+	s.Require().NoError(err, "Failed to initialize logger")
+	s.config.AWS.Region = "us-west-2"
+	s.config.Global.OrchName = "kms-test"
+	s.config.AWS.CustomerTag = "test"
+	s.runtimeState.DeploymentID = "test-deployment-id"
+	s.runtimeState.LogDir = filepath.Join(rootPath, ".logs")
+	s.tfUtility = &MockTerraformUtility{}
+	s.step = &steps_aws.KMSStep{
+		RootPath:           rootPath,
+		KeepGeneratedFiles: true,
+		TerraformUtility:   s.tfUtility,
+	}
+}
+
+func (s *KMSStepTest) TestInstallAndUninstallKMS() {
+	s.runtimeState.Action = "install"
+	s.expectTFUtiliyyCall("install")
+	rs, err := steps.GoThroughStepFunctions(s.step, &s.config, s.runtimeState)
+	if err != nil {
+		s.NoError(err)
+		return
+	}
+	fmt.Println(rs)
+
+	s.runtimeState.Action = "uninstall"
+	s.expectTFUtiliyyCall("uninstall")
+	_, err = steps.GoThroughStepFunctions(s.step, &s.config, s.runtimeState)
+	if err != nil {
+		s.NoError(err)
+	}
+}
+
+func (s *KMSStepTest) expectTFUtiliyyCall(action string) {
```

Comment:
My bad, I have a typo in VPC step test. Let's fix it here
```suggestion
func (s *KMSStepTest) expectTFUtiliyCall(action string) {
```